### ISSUE TITLE
"Biharmonic" diffusivity is a misnomer and docs are incorrect

### ISSUE THREAD

**Issue by glwagner** [CliMA/Oceananigans.jl#1279]
The docs have an incorrect description of the `AnisotropicBiharmonicDiffusivity`: 

https://clima.github.io/OceananigansDocumentation/stable/physics/turbulence_closures/#Constant-anisotropic-biharmonic-diffusivity

since the docs claim we use a horizontal biharmonic operator (dx^2 + dy^2)^2 + dz^2, despite that in our implementation:

https://github.com/CliMA/Oceananigans.jl/blob/81db22f4a26396142e8cd5b5a4c50c75790c1d50/src/TurbulenceClosures/turbulence_closure_implementations/anisotropic_biharmonic_diffusivity.jl#L64-L68

we actually use dx^4 + dy^4 + dz^4.

Our implementation is not truly biharmonic, but is instead might be described as "fourth-order hyperdiffusion". A downside of the form we use is that the amount of dissipation it provides depends on the orientation of a feature ("diagonal" gradients are dissipated less than purely horizontal or vertical gradients). An upside is that we can ensure zero fluxes on solid boundaries more easily.

We could solve this by having separate implementation of the true horizontal biharmonic operator for `AnisotropicBiharmonicDiffusivity` (still useful for horizontally-periodic problems), as well as a `FourthOrderHyperDiffusivity` that uses our current implementation. It also might make sense to have `IsotropicBiharmonicDiffusivity` since we can do triply periodic.

**Comment by francispoulin** [comment]
There are different thoughts as to what biharmonic means.

Harmonic is clearer and we all agree that the harmonic operator is the Laplacian and looks like 
```
\nabla^2 = \partial_xx + \partial_yy + \partial_zz.  
```
Biharmonic is usually defined has the harmonic operator squared, which is 
```
(\nabla^2)^2 = (\partial_xx + \partial_yy + \partial_zz)^2,
```
which involves cross terms.

What we mention in the docs is horizontal biharmonic plus vertical harmonic.  Not something that rolls off the tongue.    

The one that we currently used, is fourth-order hyperdiffusion, and makes a lot of sense in a lot of contexts.

**Comment by glwagner** [comment]
Closed by #2247
